### PR TITLE
Apply Spring application id to Azure Identity

### PR DESCRIPTION
**Pull request by saragluna**
# Description

To fix https://github.com/Azure/azure-sdk-for-java/issues/32146. 

~The test is not working because the ClientLogger gets initialized before the Spring Context Logger is ready. Need more work on the test.~

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by backwind1233** [comment]
Just curious what the final UserAgent looks like, do we have an example?

**Comment by saragluna** [comment]
> Just curious what the final UserAgent looks like, do we have an example?

```console
"User-Agent":"az-sp-id/4.6.0-beta.1 azsdk-java-azure-identity/1.8.0 (17.0.5; Mac OS X; 12.6.3)"
```

**Comment by backwind1233** [comment]
> > Just curious what the final UserAgent looks like, do we have an example?
> 
> ```
> "User-Agent":"az-sp-id/4.6.0-beta.1 azsdk-java-azure-identity/1.8.0 (17.0.5; Mac OS X; 12.6.3)"
> ```

It will also contain the other Spring Cloud Azure Identifiers, like "az-sp-sb/", right?
Like "User-Agent":"az-sp-sb/4.6.0-beta.1  az-sp-id/4.6.0-beta.1 azsdk-java-azure-identity/1.8.0 (17.0.5; Mac OS X; 12.6.3)"

**Comment by saragluna** [comment]
> > > Just curious what the final UserAgent looks like, do we have an example?
> > 
> > 
> > ```
> > "User-Agent":"az-sp-id/4.6.0-beta.1 azsdk-java-azure-identity/1.8.0 (17.0.5; Mac OS X; 12.6.3)"
> > ```
> 
> It will also contain the other Spring Cloud Azure Identifiers, like "az-sp-sb/", right? Like "User-Agent":"az-sp-sb/4.6.0-beta.1 az-sp-id/4.6.0-beta.1 azsdk-java-azure-identity/1.8.0 (17.0.5; Mac OS X; 12.6.3)"

No, it won't. The one I pasted is the final UA sent to Azure AD.

**Review by Netyyyy — APPROVED** [review]
LGTM.

**Review by backwind1233 — APPROVED** [review]
LGTM

File: sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/implementation/factory/credential/AbstractAzureCredentialBuilderFactory.java [hunk 1]
```diff
@@ -38,6 +38,7 @@
      */
     protected AbstractAzureCredentialBuilderFactory(AzureProperties azureProperties) {
         this.azureProperties = azureProperties;
+        this.setSpringIdentifier(AzureSpringIdentifier.AZURE_SPRING_IDENTITY);
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/implementation/factory/credential/AbstractAzureCredentialBuilderFactory.java:41]: Just checked the invacation of the method `setSpringIdentifier` , this identifier is wrong, please help confirm, https://github.com/Azure/azure-sdk-for-java/blob/c01270f0a3e3a9eef9b2d3c8d4b335546b3bfe9c/sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureBlobCheckpointStoreConfiguration.java#L86

**Review by stliu — APPROVED** [review]
LGTM

**Pull request merged**